### ISSUE TITLE
[dev-server-core] Disable cache

### ISSUE THREAD

**Issue by jimmy-collazos** [modernweb-dev/web#1013]
Hi, I want disable cache systen (for every files) but i can see any documentation in website.

When i see the code ([pluginTransformMiddleware.ts](https://github.com/modernweb-dev/web/blob/master/packages/dev-server-core/src/middleware/pluginTransformMiddleware.ts)), I find the mode to disable (return property `transformCache` in transform hook); but I don't know if that's good practice, because the transform hook doesn't seem like the clearest option.

This is my plugin for disable cache:
```javascript
export function noCachePlugin () {
  return {
      name: 'no-cache',
      transform: async () => ({transformCache: false})
    }
};
```
This is the good practice to disable cache?

**Comment by LarsDenBakker** [comment]
We could add a flag or option for this, but I'd like to understand a bit more about your use case and see if that's the best solution there.

**Comment by jimmy-collazos** [comment]
My case is particular because I have to maintain legacy applications (polymer 3):

I have two folders; one for the source application (`/ app`) and one for the distribution (`/dist`). I set up an observer in the `/app` directory and if a file changed, I build the application (I replace all the contents of the`/dist` directory).

_dev-server-core_ adds the file, requested by http, to the observer (it's not the same observer of `/app` directory); but that file is deleted (deleted and created) and loses the observer's reference. For this reason; server never deletes from cache and always responds from cache.

**Comment by LarsDenBakker** [comment]
I see. This is actually a bug in our file watcher then, it should notify about this deletion. 

We've been having other issues too, one solution for this would be to always doing a quick fs stat to see if the file still exists and/or has changed before serving from the cache.

**Comment by jimmy-collazos** [comment]
I opened new pull request for fix this problem:
Pleas see: https://github.com/modernweb-dev/web/pull/1015

**Comment by LarsDenBakker** [comment]
Ah, good catch!

**Comment by LarsDenBakker** [comment]
Closing this one as the original issue was solved.

**Comment by web-padawan** [comment]
Fixed by #1015
